### PR TITLE
Add --no-cleanup option to iptables-wrapper-installer.sh

### PR DESCRIPTION
**Pull request by yoheiueda**
[Cache mounts](https://docs.docker.com/build/guide/mounts/) of Docker builds are useful to share artifacts among multi-stage builds, but the cleanup mechanism of `iptables-wrapper-installer.sh` does not work well with cache mounts.

This patch adds a new option `--no-cleanup` to suppress cleanup of `iptables-wrapper` binary and `iptables-wrapper-installer.sh`.

We can use this option to build a container image with a cache mount as follows.

```
FROM golang:bookworm as builder

RUN git clone -b no-cleanup https://github.com/yoheiueda/iptables-wrappers.git /work
WORKDIR /work
RUN make BIN_DIR=/work build

FROM  debian:bookworm

RUN apt-get update && apt-get -y install iptables

RUN --mount=type=cache,target=/work,from=builder,source=/work,readonly \
	cd /work && ./iptables-wrapper-installer.sh --no-cleanup
```

### CODE REVIEW

File: iptables-wrapper-installer.sh [hunk 1]
```diff
@@ -61,7 +64,26 @@ else
...
+        ;;
+    *)
+        echo "ERROR: unknown option: $1" 1>&2
+        exit 1
+        ;;
+    esac
+    shift
+done
+
+if [ -z "$no_sanity_check" ]; then
```
Comment (danwinship) [inline iptables-wrapper-installer.sh]:
```suggestion
if [ -z "${no_sanity_check}" ]; then
```

for consistency. (Bash scripts in k8s always do it that way... I think the idea is that you _need_ the braces in some cases (like if there's text that comes directly after the variable expansion with no space between them) and it's better/more consistent to just _always_ use the braces.)

(likewise with no_cleanup below)

File: iptables-wrapper-installer.sh [hunk 2]
```diff
@@ -86,7 +108,11 @@ fi
 
 # Copy the wrapper.
 rm -f "${sbin}/iptables-wrapper"
-mv "${iptables_wrapper_path}" "${sbin}/iptables-wrapper"
+if [ -z "$no_cleanup" ]; then
+    mv "${iptables_wrapper_path}" "${sbin}/iptables-wrapper"
+else
+    cp "${iptables_wrapper_path}" "${sbin}/iptables-wrapper"
```
Comment (danwinship) [inline iptables-wrapper-installer.sh]: You could also just always do "cp" rather than "mv" here, and then rm the original in the "# Cleanup" section below

**Review by danwinship — COMMENTED** [review]
ok

File: iptables-wrapper-installer.sh (hunk 1 above)
Comment (yoheiueda) [reply iptables-wrapper-installer.sh]: Done

File: iptables-wrapper-installer.sh (hunk 2 above)
Comment (yoheiueda) [reply iptables-wrapper-installer.sh]: Done

**Comment by danwinship** [comment]
/lgtm
/approve

thanks!

**Pull request merged**